pkg/rc/replication_controller.go

```diff
@@ -264,6 +269,26 @@ func (rc *replicationController) WatchDesires(quit <-chan struct{}) <-chan error
 
 func (rc *replicationController) meetDesires(rcFields fields.RC) error {
 	rc.logger.NoFields().Infof("Handling RC update: desired replicas %d, disabled %v", rcFields.ReplicasDesired, rcFields.Disabled)
+	podID := rcFields.Manifest.ID()
+	launchableStanzas := rcFields.Manifest.GetLaunchableStanzas()
+	for launchableID, launchableStanza := range launchableStanzas {
+		artifactUrl, _, err := rc.artifactRegistry.LocationDataForLaunchable(podID, launchableID, launchableStanza)
+		if err != nil {
+			rc.logger.WithError(err).Errorln("Unable to retrieve location for launchable")
+		}
+		exists, err := rc.artifactRegistry.CheckArtifactExists(artifactUrl)
+		if err != nil {
+			rc.logger.WithError(err).Errorln("Unexpected error when checking if artifact exists")
+		}
+		if !exists {
+			if err := rc.alerter.Alert(alerting.AlertInfo{
+				Description: "This RC is missing an artifact",
+				IncidentKey: "rc_missing_artifact",
```

Comment:
i'd put the RC ID in the incident key, because pagerduty uses these to deduplicate incidents. we probably want an incident per RC that is missing an artifact

Reply:
you should probably use the `alertInfo()` function as well which puts additional helpful details into the page, like the RC's ID. You might have to modify it to take the incident key as an argument